### ISSUE TITLE
[Backport] consider backporting RWX fixes to v1.1.3

### ISSUE THREAD

**Issue by joshimoo** [longhorn/longhorn#3003]
**Is your feature request related to a problem? Please describe.**
We incorporated a couple of changes for v1.2 in regard to the RWX featureset.
We should consider backporting these for v1.1.3, there were a lot of changes in the meantime so I am creating this issue to keep track of them and for discussion purposes.

Controller logic, for terminating pods and share-manager state machine
- https://github.com/longhorn/longhorn/issues/2874 
- https://github.com/longhorn/longhorn/issues/2946

There was an attachment change here as well 
https://github.com/longhorn/longhorn-manager/pull/1010

RWX performance changes by not enforcing sync mount
- https://github.com/longhorn/longhorn/issues/2549

**Comment by joshimoo** [comment]
@khushboo-rancher once the backport PR is merged this requires a full test of all the known rwx failure cases (node failures / pod failures)

I include all the share-manager controller changes, to correctly manage the share-manager, I did not incorporate some of the newer csi changes (NodeStageVolume, etc) since they are much bigger and out of scope for the RWX failure cases.

**Comment by chanow816** [comment]
Test version : v1.1.3-rc1
Validation - **Pass**

Automation script test_rwx.py
Test result : **All rwx test case passed**

[BUG] Track terminating pods in the volume workload status #2874
Test result : **share-manager restart is not observed.**

[BUG] Deleting the only pod in a single replica ReplicaSet causes an attachment failure of the replacement pods when the volume is RWX. #2946
Test result: **share-manager restart is not observed.**

[BUG] degraded RWX performance on Ubuntu 20.04.1 LTS #2549
Test result :
**Inside of the workload** - `dd if=/dev/urandom of=<filename> count=10000 bs=1M conv=fsync status=progress`
| Longhorn Version | Size | Time taken | Avg. Speed |
|---|-----|-----|-----|
|v1.1.3-rc1| 10 Gi | 3 min 9 sec | 48.8 MB/s |

**Device** - `dd if=/dev/urandom of=/dev/longhorn/volume count=10000 bs=1M conv=fsync status=progress`
| Longhorn Version | Size | Time taken | Avg. Speed |
|---|-----|-----|-----|
|v1.1.3-rc1| 10 Gi | 3 min 34 sec | 55.3 MB/s |

cc @khushboo-rancher
